### ISSUE TITLE
Translate under module overwrite all other place

### ISSUE THREAD

**Issue by mrtuvn** [magento/magento2#37359]
### Summary

Current from my experiences, Magento process translate module as higher priority and will overwrite all other string instead replace only module scope

### Examples

Customer Login,Customer Login
Customer Login, Customer Login custom, module, Magento_Customer

The second line instead replace string in module scope will replace all other place outside of module scope.

### Proposed solution

translate module should implace into scope module only instead

Expect results
inside module customer -> text "Customer Login custom" used
other places -> will use text "Customer Login"

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Hotel** [comment]
Hello @mrtuvn,

Thanks for the report and collaboration!

We are unable to get the issue, can you please elaborate the issue more with some steps to reproduce, actual and expected result.

Thanks

**Comment by mrtuvn** [comment]
i think this case can reproduce with add translate above in theme (luma/blank) or any custom theme. You can't reproduce with clean version

Add two line below to csv in theme
Then redeploy to verify

```csv
"Customer Login","Customer Login"
"Customer Login", "Customer Login custom",module,Magento_Customer
```

Then you go to login page customer

**Comment by engcom-Hotel** [comment]
Hello @mrtuvn,

It seems that the functionality is working as expected. We have looked into the [DevDocs](https://developer.adobe.com/commerce/frontend-core/guide/translations/dictionary/), but there is no mention regarding the module-wise translation.

Please let us know in case we missed anything.

Thanks

**Comment by mrtuvn** [comment]
Yes but in the code base still use module translate but docs not mention any different or benefits of it. From my expect
when you specific module it's should only impact to module area only not impact other place not in scope of module

**Comment by engcom-Hotel** [comment]
Thanks @mrtuvn for the reply!

On this note we are confirming this issue.

Thanks

**Comment by lbajsarowicz** [comment]
No file, method, or concrete repro steps are cited here. Module-scoped translation strings overriding global ones may be the intended CSV-translation precedence rather than a defect. Can you specify which translation mechanism you're using (CSV file or inline translation) and the exact strings and modules involved?
